packages/expo-manifests/android/src/main/java/expo/modules/manifests/core/Manifest.kt

```diff
@@ -91,7 +91,7 @@ abstract class Manifest(protected val json: JSONObject) {
   abstract fun getSlug(): String?
 
   fun getDebuggerHost(): String = getExpoGoConfigRootObject()!!.require("debuggerHost")
-  fun getMainModuleName(): String = getExpoGoConfigRootObject()!!.require("mainModuleName")
+  fun getMainModuleName(): String = getExpoGoConfigRootObject()?.require("mainModuleName") ?: "main"
```

Comment:
I tried doing some searching for this on iOS and couldn't find anything. Can you elaborate in the summary about the iOS behavior this is equivalent to? (might be worth separating out this change from the snack URL changes unless they're interrelated. I'm happy to try to help dig into the `mainModuleName` ones like we did a week ago)

Reply:
Optional: Could fix this in the same way as we did the previous fix and only call this method in development mode. It looks like https://github.com/expo/expo/pull/31395 added additional calls in non-development mode.

```diff
--- a/apps/expo-go/android/expoview/src/main/java/host/exp/exponent/experience/ReactNativeActivity.kt
+++ b/apps/expo-go/android/expoview/src/main/java/host/exp/exponent/experience/ReactNativeActivity.kt
@@ -378,7 +378,11 @@ abstract class ReactNativeActivity :
     val nativeHost = ExpoGoReactNativeHost(
       application,
       instanceManagerBuilderProperties,
-      manifest!!.getMainModuleName()
+      if (delegate.isDebugModeEnabled) {
+        manifest!!.getMainModuleName()
+      } else {
+        null
+      }
     )

     val devBundleDownloadListener = ExponentDevBundleDownloadListener(progressListener)
```

I agree overall though that the behavior should match on both platform. I can't seem to find the equivalent on iOS though. 

`mainModuleName` is only available during development: https://github.com/expo/expo/blob/main/packages/%40expo/cli/src/start/server/middleware/ManifestMiddleware.ts#L286, so it seems that android may have been stricter about this constraint on the client than iOS? Either way, my approval still stands with the code as-is (making this nullable) assuming it matches iOS; Expo Go is a fairly archaic codebase and trying to do code archaeology in it usually isn't worth it.

Reply:
Here is where we provide `main` as the default on iOS https://github.com/expo/expo/blob/b2bca062a545f51961178f6ef0cf75ce20ba7d66/apps/expo-go/ios/Exponent/Kernel/ReactAppManager/EXReactAppManager.mm#L551-L570